lumicks/pylake/force_calibration/calibration_models.py

```diff
@@ -342,10 +343,27 @@ def __init__(
 
             self._passive_power_spectrum_model = passive_power_spectrum_model
 
-    def __call__(self, f, fc, diffusion_constant, *filter_params):
+    def _calculate_power_spectral_density(self, f, fc, diffusion_constant, *filter_params):
         physical_spectrum = self._passive_power_spectrum_model(f, fc, diffusion_constant)
         return physical_spectrum * self._filter(f, *filter_params)
 
+    def __call__(self, f, fc, diffusion_constant, *filter_params):
+        return self._calculate_power_spectral_density(f, fc, diffusion_constant, *filter_params)
+
+    def _alias_model(self, sample_rate, num_aliases):
+        """Include effects of aliasing into the model
+
+        Parameters
+        ----------
+        sample_rate : int
+            Sample rate in Hz
+        num_aliases : int
+            Number of aliases to take into account.
+        """
+        self._calculate_power_spectral_density = alias_spectrum(
+            self._calculate_power_spectral_density, sample_rate, num_aliases
+        )
```

Comment:
I'm a bit wary of reassigning a method like this, it seems a bit to me like introducing state to the object. I know this isn't public functionality, but I'm assuming it will be in the future? What is the intended usage?

I hesitate to suggest it, but it seems like this is an appropriate instance to use a factory.  In the `__init__()` you can just define
```
self._calculate_power_spectral_density = (
    self._passive_power_spectrum_model(f, fc, diffusion_constant) * self._filter(f, *filter_params)
)
```
and from this method here return a copy of the instance with the factory reset to the aliased version (as we do for other things).

Or if it makes more sense, have `alias=bool` as an optional argument and assign the proper function from the get go.

Reply:
I've added an explicit copy and return a new instance instead now. I don't think we need to move the construction of that function to the `__init__` for that and I'd rather keep this as a contained function if you're ok with it.

If this ends up being a full feature one day, then I would suggest moving it to the constructor, but not yet.

Note that some of the private ways in which this `CalibrationModel` are used are stateful. Note `_set_drag` for instance. We should probably move away from this, and it'd be relatively easy to do so, but this needs a little alignment with BL.

Reply:
ah sure that seems fine